### PR TITLE
handbrake: additional patch for Handbrake 1.0.7

### PR DESCRIPTION
**Pull request by kenmays**
Review additional patch fix for Handbrake 1.0.7. Tested on hrev52204 x86_64 w/haikuporter.
Ref: https://github.com/haikuports/haikuports/issues/2903

### CODE REVIEW

**Comment by extrowerk** [comment]
nice job, but you still have to bump the revision in the recipe and fix the ffmpeg constraints.

File: media-video/handbrake/patches/handbrake-1.0.7-4.patchset [hunk 1]
```diff
@@ -0,0 +1,227 @@
+From: Ken Mays <kmays2000@gmail.com>
+Date: Fri Aug 10 13:01:42 GMT 2018 
+Subject: fix for FFmpeg 4.x dependency
+
+
+--- HandBrake-1.0.7/libhb/encavcodec.c.old	2018-08-10 11:50:45.138674176 +0000
```
Comment (extrowerk) [inline media-video/handbrake/patches/handbrake-1.0.7-4.patchset]: Are you sure it applies cleanly?
Please check it again and fix it.

Comment (extrowerk) [reply media-video/handbrake/patches/handbrake-1.0.7-4.patchset]: Please, try to build stuff with haikuporter before you make any PR.
This is definetly wrong as the patches shouldn't change the filename.

Comment (kenmays) [reply media-video/handbrake/patches/handbrake-1.0.7-4.patchset]: Yes, I fixed the recipe and the patch. This is what I used to build the handbrake 1,0,7-4 package in haikuporter. I had changed the filename just for review.

File: media-video/handbrake/handbrake-1.0.7.recipe [hunk 2]
```diff
@@ -5,8 +5,8 @@ Convert from many common multimedia file formats, including unprotected \
 DVD or BluRay sources to a handful of modern output file formats."
 HOMEPAGE="http://handbrake.fr/"
 LICENSE="GNU GPL v2"
-COPYRIGHT="2003-2016 HandBrake Team"
-REVISION="3"
+COPYRIGHT="2003-2018 HandBrake Team"
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.0.7.recipe]: Revert this. You havent changed the recipe version, so this should stay unchanged,

File: media-video/handbrake/patches/handbrake-1.0.7.patchset [hunk 3]
```diff
@@ -4227,3 +4227,230 @@ index 44da58a..0000000
...
+@@ -341,7 +341,7 @@
+             fclose( pv->file );
+             pv->file = NULL;
+ 
+-            context->flags    |= CODEC_FLAG_PASS2;
++            context->flags    |= AV_CODEC_FLAG_PASS2;
+             context->stats_in  = log;
+         }
+     }
+--- HandBrake-1.0.7/libhb/muxavformat.c.old	2018-08-10 11:54:00.651165696 +0000
```
Comment (extrowerk) [inline media-video/handbrake/patches/handbrake-1.0.7.patchset]: Where does the "old" postfix comes from?

File: media-video/handbrake/handbrake-1.0.7.recipe (hunk 2 above)
Comment (extrowerk) [reply media-video/handbrake/handbrake-1.0.7.recipe]: btw there is a new version here (1.1.1) you should bump the version.

File: media-video/handbrake/patches/handbrake-1.0.7.patchset (hunk 3 above)
Comment (alaviss) [reply media-video/handbrake/patches/handbrake-1.0.7.patchset]: probably they patched it by moving files around, not via the built in HaikuPorter git repo. It won't cause any harm, just not the preferred way

**Comment by kenmays** [comment]
Changes made. Test on your buildbot for review.

**Comment by kenmays** [comment]
Added Handbrake 1.1.1

**Comment by kenmays** [comment]
Closing this one. Pushed to Handbrake 1.1.1 update.

**Pull request closed without merging**